api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -535,14 +535,13 @@ public void purgeUser(User user, boolean cascade) throws APIException {
 	 */
 	private void checkPrivileges(Role role) {
 		Collection<Privilege> privileges = role.getPrivileges();
-		
-		if (privileges != null) {
-			for (Privilege p : privileges) {
-				if (!Context.hasPrivilege(p.getPrivilege())) {
-					throw new APIAuthenticationException("Privilege required: " + p);
-				}
-			}
-		}
+
+		Optional.ofNullable(role.getPrivileges())
+		.map(p -> p.stream().filter(pr -> !Context.hasPrivilege(pr.getPrivilege())).collect(
+			Collectors.toSet())).ifPresent(missing -> {
+		throw new APIAuthenticationException("privileges required: " + missing.stream().map(
```

Comment:
This should at least be "Privileges required", but it might be better to do this in a localisable way as you did for the other ticket.